### PR TITLE
Saghi cms unique kmers

### PR DESCRIPTION
**Pull request by Gnojoomi**
I created two strings of sequences with some overlapping Kmers and I used Khmer to find unique Kmers to one of the sequences. 
- [x] Is it mergeable?
- [x] `make test` Did it pass the tests?
- <strike>[ ] `make clean diff-cover` If it introduces new functionality in
  `scripts/` is it tested?</strike>
- [x] `make format diff_pylint_report cppcheck doc pydocstyle` Is it well
  formatted?
- <strike>[ ] Did it change the command-line interface? Only backwards-compatible
  additions are allowed without a major version increment. Changing file
  formats also requires a major version number increment.</strike>
- <strike>[ ] For substantial changes or changes to the command-line interface, is it
  documented in `CHANGELOG.md`? See [keepachangelog](http://keepachangelog.com/)
  for more details.</strike>
- <strike>[ ] Was a spellchecker run on the source code and documentation after
  changes were made?</strike>
- <strike>[ ] Do the changes respect streaming IO? (Are they
  tested for streaming IO?)</strike>

### CODE REVIEW

File: examples/python-api/myownbloom.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env python
+import khmer
+
```
Comment (standage) [inline examples/python-api/myownbloom.py]: Similar to the other examples, a short summary at the top would be helpful.

File: examples/python-api/myownbloom.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+bf =khmer.Nodetable(ksize, target_table_size, num_tables);
+bf.consume(d1);
+
+cms=khmer.Counttable(ksize, target_table_size, num_tables);
+
+for kmer in cms.get_kmers(d2): #each kmer in d2
+	if bf.get(kmer) == 0:#if not in d1=
+		cms.consume(kmer)
+assert cms.get('CCAAA')
```
Comment (standage) [inline examples/python-api/myownbloom.py]: The `get` command returns an integer. This assert statement will fail if the `get` function returns a false value (0 in the case of integers) and will pass if it returns a non-false value (any other integer). I'd suggest being more explicit and showing exactly what you're testing for: `assert cmd.get('CCAAA') > 0`.

File: examples/python-api/myownbloom.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+bf.consume(d1);
+
+cms=khmer.Counttable(ksize, target_table_size, num_tables);
+
+for kmer in cms.get_kmers(d2): #each kmer in d2
+	if bf.get(kmer) == 0:#if not in d1=
+		cms.consume(kmer)
+assert cms.get('CCAAA') 
+
+assert bf.get('ACCGA')
```
Comment (standage) [inline examples/python-api/myownbloom.py]: This command is testing whether the k-mer is in the mask. I'm not sure that's a helpful test.

What *would* be helpful is to find a k-mer that is present in both d1 and d2 and make sure it's NOT in cms (that is `assert cms.get(thekmer) == 0`).

**Comment by standage** [comment]
Also, I'm going to cross out some of the checklist items that aren't relevant.

**Comment by standage** [comment]
Another comment: the file is named `myownbloom.py`, but this is something we actually want to integrate into the official code base. A file name that is more descriptive (something like `load-mask.py` or similar) would be more appropriate.

**Comment by ctb** [comment]
you can run 'make pep8', note

**Comment by standage** [comment]
> you can run 'make pep8', note

I don't think pep8 tests the example scripts though. Probably should :-)

**Comment by standage** [comment]
In 588b6490 I changed the Makefile so that `make pep8` checks Python example files too.

File: examples/python-api/mask.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+ksize = 5
+target_table_size = 5e8
+num_tables = 4
+bf = khmer.Nodetable(ksize, target_table_size, num_tables)
+bf.consume(d1)
+cms = khmer.Counttable(ksize, target_table_size, num_tables)
+for kmer in cms.get_kmers(d2):
+    if bf.get(kmer) == 0:
+        cms.consume(kmer)
+assert cms.get('CATTA') == 0
```
Comment (standage) [inline examples/python-api/mask.py]: See https://github.com/dib-lab/khmer/pull/1832/files#diff-361d677c398f6c5b3d85f554ae9213f9 for some additional assertions.

**Comment by standage** [comment]
In ea0f8ce0 I made sure `mask.py` is run when the user invokes `make py-demos`, and I changed the permissions of the file to make it executable.

File: examples/python-api/mask.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env python
+# The following program uses khmer to
+# find unique kmers to only one sequence.
+import khmer
+d1 = "ATGTACGGGCATTACGATTACCGATGTAG"
+d2 = "ATGACCAAACTCATTACGATTAGATATAG"
+ksize = 5
+target_table_size = 5e8
```
Comment (standage) [inline examples/python-api/mask.py]: Can use much less here. Like 5e5 or 5e6.

File: examples/python-api/mask.py [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+bf = khmer.Nodetable(ksize, target_table_size, num_tables)
+bf.consume(d1)
+cms = khmer.Counttable(ksize, target_table_size, num_tables)
+for kmer in cms.get_kmers(d2):
+    if bf.get(kmer) == 0:
+        cms.consume(kmer)
+
+# If kmer is in both sequences it should not be in cms but in bf
+assert cms.get('CATTA') == 0
+assert bf.get('CATTA') == 0
```
Comment (standage) [inline examples/python-api/mask.py]: As discussed, this and line 22 are problematic.

**Comment by standage** [comment]
Ok, everything looks good now. I'll push the shiny "Merge" button once the continuous integration tests pass.

**Comment by betatim** [comment]
Hit merge for you @standage, have you escaped the airline re-routing maze yet? 🛫

**Comment by standage** [comment]
Thanks @betatim! Yes it all turned out fine.

And thanks @Gnojoomi and @jchow32 for your first contribution to khmer!

**Pull request merged**